crates/bevy_winit/src/lib.rs

```diff
@@ -86,41 +85,31 @@ impl Plugin for WinitPlugin {
 
         // This is needed because the features checked in the inner
         // block might be enabled on other platforms than linux.
-        #[cfg(target_os = "linux")]
+        #[cfg(all(target_os = "linux", feature = "x11"))]
         {
-            #[cfg(feature = "x11")]
-            {
-                use winit::platform::x11::EventLoopBuilderExtX11;
+            use winit::platform::x11::EventLoopBuilderExtX11;
 
-                // This allows a Bevy app to be started and ran outside of the main thread.
-                // A use case for this is to allow external applications to spawn a thread
-                // which runs a Bevy app without requiring the Bevy app to need to reside on
-                // the main thread, which can be problematic.
-                event_loop_builder.with_any_thread(self.run_on_any_thread);
-            }
-
-            #[cfg(feature = "wayland")]
-            {
-                use winit::platform::wayland::EventLoopBuilderExtWayland;
-                event_loop_builder.with_any_thread(self.run_on_any_thread);
-            }
+            // This allows a Bevy app to be started and ran outside of the main thread.
+            // A use case for this is to allow external applications to spawn a thread
+            // which runs a Bevy app without requiring the Bevy app to need to reside on
+            // the main thread, which can be problematic.
+            event_loop_builder.with_any_thread(self.run_on_any_thread);
+        }
+        #[cfg(all(target_os = "linux", feature = "wayland"))]
+        {
+            use winit::platform::wayland::EventLoopBuilderExtWayland;
+            event_loop_builder.with_any_thread(self.run_on_any_thread);
         }
-
         #[cfg(target_os = "windows")]
         {
             use winit::platform::windows::EventLoopBuilderExtWindows;
             event_loop_builder.with_any_thread(self.run_on_any_thread);
         }
-
         #[cfg(target_os = "android")]
         {
             use winit::platform::android::EventLoopBuilderExtAndroid;
-            event_loop_builder.with_android_app(
-                ANDROID_APP
-                    .get()
-                    .expect("Bevy must be setup with the #[bevy_main] macro on Android")
-                    .clone(),
-            );
+            let msg = "Bevy must be setup with the #[bevy_main] macro on Android";
+            event_loop_builder.with_android_app(ANDROID_APP.get().expect(msg).clone());
```

Comment:
I don't really understand why you are separating the `msg` into a separate variable. If you wanted to reduce indentation I think this will work better:
```suggestion
            let app = ANDROID_APP
                .get()
                .expect("Bevy must be setup with the #[bevy_main] macro on Android")
                .clone();
            event_loop_builder.with_android_app(app);
```

Reply:
It's just a personal style choice. I prefer to avoid expanding the method chain into several lines, it occupies vertical space and you can see less things on screen at the same time.

But it's purely personal preference. If the community prefers the expanded version, I'll switch back.

Reply:
Alright that's fine by me.